src/Output/MultipleFileOutput.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace Spatie\TypeScriptTransformer\Output;
+
+use Spatie\TypeScriptTransformer\TypeScriptTransformerConfig;
+
+class MultipleFileOutput extends Output
+{
+    public function __construct(protected TypeScriptTransformerConfig $config)
+    {
+        $path = $config->getOutputDestination();
+        $this->path = is_dir($path) ? rtrim($path, '/') : pathinfo($path, PATHINFO_DIRNAME);
+        $this->output = [];
+
+    }
+
+    public function append(string $typeScript, string $classPath = null): void
+    {
+        $pathName = ltrim(str_replace('\\', '/', $classPath), '/');
+
+        $this->output[$pathName] = $typeScript;
+    }
+
+    public function writeOut($fileType): array
+    {
+        $this->clearPrevious($fileType);
+
+        foreach ($this->output as $path => $contents) {
+            if ($contents === '') {
+                continue;
+            }
+
+            $path = "$this->path/$path.{$fileType}";
```

Comment:
"{$this->path}/{$path}.{$fileType}"

maybe rename `$this->path` to `$rootPath`? `{$this->path}/{$path}` is kinda confusing